packages/ra-core/src/controller/field/useReference.ts

```diff
@@ -50,7 +52,8 @@ export interface UseReferenceProps {
  * @param {Object} option
  * @param {boolean} option.allowEmpty do we allow for no referenced record (default to false)
  * @param {string} option.basePath basepath to current resource
- * @param {string | false} option.linkType The type of the link toward the referenced record. edit, show of false for no link (default to edit)
+ * @param {string | false | linkTypeFunction} option.linkType The type of the link toward the referenced record. 'edit', 'show' or
+ * false for no link (default to edit). Alternatively a function that returns a string
```

Comment:
Please don't break the comment line, I don't think all jsdoc readers handle it correctly.